docs/src/aleph_bft_api.md

```diff
@@ -67,9 +67,43 @@ pub trait KeyBox: Index + Clone + Send {
 
 A typical implementation of KeyBox would be a collection of `N` public keys, an index `i` and a single private key corresponding to the public key number `i`. The meaning of `sign` is then to produce a signature using the given private key, and `verify(msg, s, j)` is to verify whether the signature `s` under the message `msg` is correct with respect to the public key of the `j`th node.
 
+#### 3.1.3 IntoBackup, IntoReader.
+
+The `IntoBackup` and `IntoReader` traits are used for creating backups of Units created in a session. This is a part of crash recovery. Units created are needed for member to recover after crash during a session for Aleph to be BFT. This means that user needs to provide two traits `IntoBackup` and `IntoReader` that are used for storing and reading Unit that are created by member. At first (without any crash) `IntoReader` should return nothing. After crash it should contain all data that was stored before in this session.
+
+`IntoBackup` trait upon being consumed by `into_backup` method should return a struct implementing `Backup<T>` trait. It should implement for all `T: Codec + Send + Sync` a way of buckuping a data generated during session. **If member crashed `Backup<T>` should append data that was saved before the crash.**
```

Comment:
Is it that hard? It should append to the data that was passed in `Read`. It does not sound that hard if we already have this data somewhere. Unless you mean that it's not so easy, to have it saved?

Reply:
Depending on what you mean by "crash", if the whole machine crashes you cannot really guarantee anything. But I guess this is kinda clear from context...?

Reply:
Yep, I think of it as if you have something saved, you should append it. It's not it should append all data generated before crash. So maybe "should append data that was saved despite the crash"?

Reply:
Hmm, if I understand correctly what property you are trying to ensure perhaps something like "writing bytes should block until it's saved in a way that should be persistent, e.g. on a disk"? I think this is the actual property we want, since it ensures (as much as is practically possible) that stuff is saved before it's send to network.

Reply:
Hmmm, about this one I thought a little bit differently. We ask to implement `Write` so it has two methods. `write` and `flush`. So I guess the user should implement it in a typical way and we use flush to make sure that all is saved

Reply:
`Write` doesn't necessarily write to disk on `flush` (it's implemented for many interfaces and they can do pretty arbitrary things) so the actual requirement should be that `flush` blocks until the written stuff is persistently saved.

Reply:
How do you think we can emphasize this requirement? Do you think mentioning it here is enough?

Reply:
Yeah, I think just mentioning it here is enough, especially since this is kinda the default implementation in many cases.

Reply:
I think the second bold sentence is now redundant.